mocktracer/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+include_directories(include)
+include_directories(3rd_party/base64/include)
+
+set(SRCS 3rd_party/base64/src/base64.cpp
+         src/mock_span_context.cpp
+         src/mock_span.cpp
+         src/in_memory_recorder.cpp
+         src/propagation.cpp
+         src/tracer.cpp)
+
+if (BUILD_SHARED_LIBS)
+  add_library(opentracing_mocktracer SHARED ${SRCS})
+  target_include_directories(opentracing_mocktracer INTERFACE "$<INSTALL_INTERFACE:include/>")
+  set_target_properties(opentracing_mocktracer PROPERTIES VERSION ${OPENTRACING_VERSION_STRING}
+                                               SOVERSION ${OPENTRACING_VERSION_MAJOR})
+  target_link_libraries(opentracing_mocktracer opentracing)
+  install(TARGETS opentracing_mocktracer EXPORT OpenTracingTargets
+          LIBRARY DESTINATION lib
+          ARCHIVE DESTINATION lib)
+endif(BUILD_SHARED_LIBS)
```

Comment:
Not asking you to change this, just an FYI, the CMake docs seem to be wrong about this point. The expression need not/should not be included in `endif` arguments. None of the CMake modules follow that pattern.